### PR TITLE
Ge05

### PR DESCRIPTION
**Pull request by jivp-odoo**
With this module we automate the way in which warehouses are selected.

### CODE REVIEW

File: ge05_team01/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                    This Module is used to computed field based off of the delivery address. 
+                    Developer: [JIVP]
+                    Task ID: 3426696
+                    Link to task: https://www.odoo.com/web#id=3426696&menu_id=4720&cids=17&action=4665&active_id=3426716&model=project.task&view_type=form
+                    """,
+    "version": "1.0.0",
+    "category": "Custom Development",
+    "license": "OPL-1",
+    "depends": ["motorcycle_registry", "sale"],
+    "data": [],
```
Comment (gecm-odoo) [inline ge05_team01/__manifest__.py:14]: Remove empty tag

File: ge05_team01/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "author": "team-01",
+    "website": "www.odoo.com",
+    "installable": True
```
Comment (gecm-odoo) [inline ge05_team01/__manifest__.py:17]: You do not need the installable tag

File: ge05_team01/models/calculate_distance.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from odoo import fields, models, api
+
+san_francisco = ["California", "Oregon", "Washington",
+                 "Nevada", "Arizona", "Utah", "Idaho", "Montana", "Wyoming", "Colorado", "New Mexico", "Sonora", "Sinaloa", "Nayarit", "Baja California", "Baja California Sur"]
+buffalo = ["New York", "Pennsylvania", "Ohio", "Michigan", "Indiana", "West Virginia", "Vermont",
```
Comment (gecm-odoo) [inline ge05_team01/models/calculate_distance.py:5]: Using lists definitely works, however, can you come up with a solution using the odoo fields? That way we can store them in the DB and in the future, maybe add to the lists without messing with code.

File: ge05_team01/models/calculate_distance.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    _inherit = "sale.order"
+
+    @api.depends("user_id", "company_id", "partner_id")
+    def _compute_warehouse_id(self):
+        our_warehouses = [
+            (san_francisco, "San Francisco Dealership"),
+            (buffalo, "Buffalo Dealership"),
+        ]
+
+        for sales in self:
```
Comment (gecm-odoo) [inline ge05_team01/models/calculate_distance.py:19]: for order in self

**Review by gecm-odoo — CHANGES_REQUESTED** [review]
Check out the computed fields documentation https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#computed-fields

File: ge05_team01/models/calculate_distance.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                 "Nevada", "Arizona", "Utah", "Idaho", "Montana", "Wyoming", "Colorado", "New Mexico", "Sonora", "Sinaloa", "Nayarit", "Baja California", "Baja California Sur"]
+buffalo = ["New York", "Pennsylvania", "Ohio", "Michigan", "Indiana", "West Virginia", "Vermont",
+           "New Hampshire", "Maine", "Massachusetts", "Rhode Island", "Connecticut", "Delaware", "Maryland", "Tamaulipas", "Nuevo Leon", "Coahuila", "Chihuahua", "Durango", "Zacatecas", "San Luis Potosi", "Aguascalientes", "Jalisco"]
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("user_id", "company_id", "partner_id")
+    def _compute_warehouse_id(self):
```
[the remark this answers is not in the document]
Comment (jivp-odoo) [reply ge05_team01/models/calculate_distance.py:13]: Hi Gerry,

In odoo/addons/sale_stoke/models/sale_order.py the warehouse_id field is predefined as compute.

`warehouse_id = fields.Many2one(
        'stock.warehouse', string='Warehouse', required=True,
        compute='_compute_warehouse_id', store=True, readonly=False, precompute=True,
        states={'sale': [('readonly', True)], 'done': [('readonly', True)], 'cancel': [('readonly', False)]},
        check_company=True)`

Do i need to redefine this?

File: ge05_team01/models/calculate_distance.py (hunk 3 above)
Comment (jivp-odoo) [reply ge05_team01/models/calculate_distance.py:5]: This is something that i'll try to implement in my free time hahaha by the moment we are going to start with task 7, what do you think master? x)

File: ge05_team01/models/calculate_distance.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (gecm-odoo) [reply ge05_team01/models/calculate_distance.py:13]: My bad, you are correct

**Review by gecm-odoo — APPROVED** [review]

**Pull request merged**